crates/derive/src/stages/batch_queue.rs

```diff
@@ -343,7 +343,11 @@ where
         match batch {
             Batch::Single(sb) => Ok(sb),
             Batch::Span(sb) => {
-                let batches = sb.get_singular_batches(&self.l1_blocks, parent);
+                let batches = sb.get_singular_batches(&self.l1_blocks, parent).map_err(|e| {
+                    StageError::Custom(anyhow!(
+                        "Could not get singular batches from span batch: {e}"
+                    ))
+                })?;
```

Comment:
Can we add a test that the batch queue bubbles up this error if it fails to get singular batches? Not sure how easy it would be to construct the span batch such that the singular batch decoding fails here.

Reply:
Going to follow up on this, but should be able to do this with some improved test utilities. Would like something similar to [`RandomRawSpanBatch`](https://github.com/ethereum-optimism/optimism/blob/8dd17a7b114a7c25505cd2e15ce4e3d0f7e3f7c1/op-node/rollup/derive/batch_test.go#L21) from the `op-node` test suite.

All of the `alloy` types we use have `Arbitrary` trait implementations with the `arbitrary` feature, which we can activate in the `test` profile. Would need to derive `Arbitrary` for our own types and add an `arbitrary` feature flag to `kona-derive`, but will be well worth :D